### PR TITLE
Improve SSL pretraining robustness for all-padding windows

### PR DESCRIPTION
**Pull request by sappho192**
### Motivation
- The SSL pretraining pipeline could crash when encountering rare windows with zero valid (non-padding) notes because losses were computed over empty tensors. 
- The change aims to make pretraining robust so long-running SSL jobs on GiantMIDI do not fail on these edge cases and to fail fast on invalid `mask_ratio` inputs.

### Description
- Added validation for `mask_ratio` in `v3/mvi_v3/data/pretrain_dataset.py` to raise a `ValueError` when outside `[0, 1]`.
- Made `PretrainWindowDataset` safe for windows with zero valid notes by returning an empty mask instead of selecting indices that would produce errors.
- Updated `v3/mvi_v3/training/pretrain_engine.py` (`run_pretrain_epoch`) to handle `n_masked == 0` by producing a graph-connected zero loss instead of computing CE/MSE on empty tensors.
- Added focused regression tests in `v3/tests/test_pretrain_ssl_edge_cases.py` covering all-padding windows, zero-masked-position training behavior, and `mask_ratio` validation.

### Testing
- Ran the targeted test module with `uv run pytest -q tests/test_pretrain_ssl_edge_cases.py`, which passed: `3 passed, 1 warning`.
- Running the full test suite with `cd v3 && pytest -q` aborted during collection because `numpy` was not installed in the outer test environment, producing import errors (collection failure), unrelated to the changes here.

------
[Codex Task](https://chatgpt.com/codex/tasks/task_e_69b50bec2384832aa8f90a2cf6042ac8)

### CODE REVIEW

**Comment by sappho192** [comment]
pretrain_dataset.py - 좋음
	∙	mask_ratio 범위 검증과 n_valid == 0 분기 처리가 정확합니다.
	∙	원본 코드에서 max(1, int(0 * 0.15)) = 1이 되어 빈 valid_indices에서 IndexError가 발생하는 문제를 깔끔하게 해결했습니다.
pretrain_engine.py - 좋음
	∙	pitch_logits.sum() * 0.0으로 graph-connected zero loss를 생성하는 것은 autograd 그래프를 유지하면서 gradient 흐름을 보장하는 표준 패턴입니다.
	∙	gradient accumulation과의 상호작용도 올바릅니다.
test_pretrain_ssl_edge_cases.py - 대체로 좋음
	∙	3개의 테스트 케이스가 변경사항을 적절히 검증합니다.
개선 제안 (선택사항)
	1.	(Style) test_mask_ratio_validation에서 try/except 대신 pytest.raises(ValueError) 컨텍스트 매니저를 사용하면 더 간결하고 관용적입니다:
```
def test_mask_ratio_validation():
    with pytest.raises(ValueError, match="mask_ratio"):
        PretrainWindowDataset([_window(False)], config=BaselineConfig(), mask_ratio=1.5)
```
	2.	(참고, PR 범위 밖) mask_ratio=0.0이 허용되지만 n_mask = max(1, ...) 로직으로 인해 항상 최소 1개의 노트가 마스킹됩니다. 향후 mask_ratio=0.0의 의미적 기대(마스킹 없음)와 실제 동작의 불일치를 정리하면 좋겠습니다.
